docs/source/training_tutorials/sft_lora_finetune_llm.mdx

```diff
@@ -0,0 +1,424 @@
+tensor_parallel_size = 2
+
+dataset = load_dataset("databricks/databricks-dolly-15k", split="train")
+
+model_id = "meta-llama/Meta-Llama-3-8B"
+
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+tokenizer.pad_token = tokenizer.eos_token
+
+with lazy_load_for_parallelism(tensor_parallel_size=tensor_parallel_size):
+    model = AutoModelForCausalLM.from_pretrained(model_id)
+
+config = LoraConfig(
+    r=16,
+    lora_alpha=16,
+    lora_dropout=0.05,
+    target_modules=[
+        "q_proj",
+        "gate_proj",
+        "v_proj",
+        "o_proj",
+        "k_proj",
+        "up_proj",
+        "down_proj"
+    ],
+    bias="none",
+    task_type="CAUSAL_LM",
+)
+
+# training_args is an instance of NeuronTrainingArguments
+args = training_args.to_dict()
+sft_config = NeuronSFTConfig(
+    max_seq_length=1024,
+    packing=False,
+    **args,
+)
+
+trainer = NeuronSFTTrainer(
+    args=sft_config,
+    model=model,
+    peft_config=config,
+    tokenizer=tokenizer,
+    train_dataset=dataset,
+    formatting_func=format_dolly,
+)
+
+# Start training
+trainer.train()
+
+trainer.save_model()  # Saves the tokenizer too for easy upload
+```
+
+The key points here are:
+
+- We use the `lazy_load_for_parallelism` context manager to lazily load the model. This will not load the full model weights on each worker, but instead only load the required weights (sharded or full). **This is much more memory efficient, and often mandatory to use.**
+- We define a `LoraConfig` that specifies which layers should have adapters, and the hyperparameters for theses adapters.
+- We create a [`~optimum.neuron.NeuronSFTConfig`] from regular `NeuronTrainingArguments`. Here we specify that we do not want to pack our examples, and that the max sequence length should be `1024`, meaning that every example will be either padded or truncated to a length of `1024`.
+- We use the [`~optimum.neuron.NeuronSFTTrainer`] to perform training. It will take the lazily loaded model, along with `lora_config`, `sft_config` and `format_dolly` and prepare the dataset and model for supervised fine-tuning.
+
+## 4. Launch Training 
+
+We prepared a script called [sft_lora_finetune_llm.py](https://github.com/huggingface/optimum-neuron/blob/main/docs/source/training_tutorials/lora_finetune_llm.py) summing up everything mentioned in this tutorial.
+
+PyTorch Neuron uses `torch_xla`. It evaluates operations lazily during execution of the training loops, which means it builds a symbolic graph in the background and the graph is executed on the hardware only when the tensor is printed, transfered to CPU, or `xm.mark_step()` is called. During execution, multiple graphs can be built depending on control-flow and it can take time to compile each graph sequentially. To alleviate that, the Neuron SDK provides `neuron_parallel_compile`, a tool which performs a fast trial run that builds all the graphs and compile them in parallel. This step is usually called precompilation.
```

Comment:
```suggestion
PyTorch Neuron uses `torch_xla`. It evaluates operations lazily during the execution of the training loops, which means it builds a symbolic graph in the background, and the graph is executed on the hardware only when the tensor is printed, transferred to CPU, or when `xm.mark_step()` is called. During execution, multiple graphs can be built depending on control-flow, and it can take time to compile each graph sequentially. To alleviate that, the Neuron SDK provides `neuron_parallel_compile`, a tool which performs a fast trial run that builds all the graphs and compile them in parallel. This step is usually called precompilation.
```